cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/PatConfig.java

```diff
@@ -45,19 +48,34 @@ public String getPasswordName() {
 
   @Nullable
   public String getUsername() {
-    return username;
+    // If username is not specified, use the default username.
+    return username == null || username.isEmpty() ? DEFAULT_USERNAME : username;
   }
 
   /**
    * Validate PatConfig.
-
+   *
    * @return a collection of {@link RepositoryValidationFailure}.
    */
-  public Collection<RepositoryValidationFailure> validate() {
+  public Collection<RepositoryValidationFailure> validate(Provider provider) {
     Collection<RepositoryValidationFailure> failures = new ArrayList<>();
     if (passwordName == null || passwordName.equals("")) {
-      failures.add(new RepositoryValidationFailure("'passwordName' must be specified in 'patConfig'."));
+      failures.add(
+          new RepositoryValidationFailure("'passwordName' must be specified in 'patConfig'."));
     }
+    if (getUsername() == null) {
+      failures.add(
+          new RepositoryValidationFailure("'username' must be specified in 'patConfig'."));
+    }
+
+    if (provider == Provider.BITBUCKET_CLOUD || provider == Provider.BITBUCKET_SERVER) {
+      if (!getUsername().equals("x-token-auth")) {
```

Comment:
generally, in java we use constants before in `equals`.

For example,

`!DEFAULT_USERNAME.equals(getUsername)`